src/vnsw/agent/services/dhcp_handler_base.cc

```diff
@@ -355,17 +355,11 @@ uint16_t DhcpHandlerBase::AddByteCompressedNameOption(uint32_t option,
 
 uint16_t DhcpHandlerBase::AddCompressedName(uint16_t opt_len,
                                             const std::string &input) {
-#ifndef _WINDOWS
-    uint8_t name[input.size() * 2 + 2];
-#else
-	uint8_t *name = new uint8_t[input.size() * 2 + 2];
-#endif
+    uint8_t *name = new uint8_t[input.size() * 2 + 2];
```

Comment:
I think we should use smart pointers here:
```
std::unique_ptr<uint8_t[]> name(new uint8_t[input.size() * 2 + 2]);
```

Reply:
I agree. Another option is using `std::vector<uint8_t>`.